cluster_management/src/main/java/com/pinterest/rocksplicator/task/BackupTaskFactory.java

```diff
@@ -0,0 +1,121 @@
+/// Copyright 2017 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+//
+// @author kangnanli (kangnanli@pinterest.com)
+//
+
+package com.pinterest.rocksplicator.task;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.helix.task.JobConfig;
+import org.apache.helix.task.Task;
+import org.apache.helix.task.TaskCallbackContext;
+import org.apache.helix.task.TaskConfig;
+import org.apache.helix.task.TaskFactory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * An implementation of {@link TaskFactory} that create Tasks upon backup-job launching.
+ *
+ * This class takes in {@param adminPort}, {@param cluster}, {@param zkClient} from
+ * {@link com.pinterest.rocksplicator.Participant} used to identify job (tasks) belonging.
+ * {@param zkClient} used to access zNode from zk cluster
+ *
+ * Upon Job creation, take the {@link JobConfig} instantiation time from {@link TaskCallbackContext}
+ * as a universal version for tasks of the job.
+ * jobVersion == resourceVersion
+ *
+ * Upon Task creation, take in params from JobConfig which could be passed at job creation;
+ * JobCommandConfigMap: (optional)BACKUP_LIMIT_MBS, specify backup limit by mbs;
+ * Upon Task execution, store resourceVersion, resourceCluster, jobVersion in userStore at Job level
+ */
+public class BackupTaskFactory implements TaskFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(BackupTaskFactory.class);
+
+  private final String host;
+  private final int adminPort;
+  private final String cluster;
+  private CuratorFramework zkClient;
+
+  private static final int DEFAULT_BACKUP_LIMIT_MBS = 64;
+
+  public BackupTaskFactory(String host, int adminPort, String zkConnectString, String cluster) {
+    this.host = host;
```

Comment:
Do we really need it?

Reply:
We may want to remove host from Task and TaskFactory as it's only used in a logging. Local host name is not that useful in a log.